hs_core/management/commands/list_published_by_year.py

```diff
@@ -37,47 +40,74 @@ def add_arguments(self, parser):
             help='limit to resources owned by specific user'
         )
 
+        parser.add_argument(
+            '--csv',
+            action='store_true',  # True for presence, False for absence
+            dest='csv',
+            help='export as csv'
+        )
+
     def handle(self, *args, **options):
         days = options['days']
         resources = BaseResource.objects.filter(raccess__published=True)
         owner = options['owned_by']
         res_type = options['type']
+        csv = options['csv']
+        year = options['year']
+        file_name = ''
 
         if owner is not None:
             try:
                 owner = User.objects.get(username=owner)
                 resources.filter(r2urp__user=owner,
                                  r2urp__privilege=PrivilegeCodes.OWNER)
+                if csv:
+                    file_name += f"_owner={owner}"
             except ObjectDoesNotExist:
                 print(f"User matching {owner} not found")
 
         if res_type is not None:
             if res_type in ["CompositeResource", "CollectionResource"]:
                 resources.filter(resource_type=res_type)
+                file_name += f"_restype={res_type}"
             else:
                 print(f"Type {res_type} is not supported. Must be 'CompositeResource' or 'CollectionResource'")
 
+        if csv:
+            file_name += f"_run={timezone.now()}"
+            if year:
+                file_name += f"_year={year}"
+
         resources = resources.order_by(F('updated').asc(nulls_first=True))
+        site_url = hydroshare.utils.current_site_url()
+
+        if csv:
+            self.csvfile = open(f'published_resources{file_name}.csv', 'w')
+            self.writer = csv_module.writer(self.csvfile)
 
         for resource in resources:
             pub_date = self.get_publication_date(resource)
-            if options['year']:
-                if pub_date.year != int(options['year']):
+            if year:
+                if pub_date.year != int(year):
                     continue
             if days:
                 cuttoff_time = timezone.now() - timedelta(days)
                 if not pub_date >= cuttoff_time:
                     continue
-            self.print_resource(resource, pub_date)
+            if csv:
+                self.build_csv(resource, pub_date, site_url)
+            else:
+                self.print_resource(resource, pub_date, site_url)
+        if csv:
+            self.write_csv(self.writer)
 
     def get_publication_date(self, resource):
         published_date = resource.metadata.dates.filter(type="published").first()
         if not published_date:
             print(f"Publication date not found for {resource.short_id}")
         return published_date.start_date
```

Comment:
If we are expecting not to find the `publication_date`, should we return `None`? Otherwise, line 108 we will raise exception.

Reply:
good call thanks